tests/bluechi_test/systemctl.py

```diff
@@ -31,9 +32,29 @@ def start_unit(self, unit_name: str, check_result: bool = True, expected_result:
             -> Tuple[Optional[int], Union[Iterator[bytes], Any, Tuple[bytes, bytes]]]:
         return self._do_operation_on_unit(unit_name, "start", check_result, expected_result)
 
-    def stop_unit(self, unit_name: str, check_result: bool = True, expected_result: int = 0)  \
+    def stop_unit(self,
+                  unit_name: str,
+                  check_result: bool = True,
+                  expected_result: int = 0,
+                  operation_results: Set[str] = ("success", "exit-code"))  \
             -> Tuple[Optional[int], Union[Iterator[bytes], Any, Tuple[bytes, bytes]]]:
-        return self._do_operation_on_unit(unit_name, "stop", check_result, expected_result)
+        result, output = self._do_operation_on_unit(unit_name, "stop", check_result, expected_result)
+
+        if check_result:
+            time.sleep(1)
+            # It's not enough to check result for systemctl stop, it will return failure only if stopping unit is
+            # prohibited or unit doesn't exist, so we need to check also operation result
+            op_res, op_out = self._do_operation_on_unit(unit_name, 'show --property="Result"', check_result, 0)
+            if op_res != 0:
+                raise Exception(
+                    f"Failed to fetch stop operation result for service {unit_name} with {op_res}: {op_out}")
+            op_res_value = op_out.replace("Result=", "")
+            if op_res_value not in operation_results:
+                raise Exception(
+                    f"Operation stop on service {unit_name} finished with result {op_res_value}"
+                    " (expected {operation_results})")
```

Comment:
NIT: The second string in the new line also needs an `f` like so `f" (expected {operation_results})")

Reply:
Done